solidity/contracts/vortex/VortexBurner.sol

```diff
@@ -0,0 +1,522 @@
+     *
+     * @return total amount of the burned governance tokens
+     */
+    function totalBurnedAmount() external view returns (uint256) {
+        return _totalBurnedAmount;
+    }
+
+    /**
+     * @dev converts the provided tokens to governance tokens and burns them
+     *
+     * @param tokens the tokens to convert
+     */
+    function vortex(IERC20[] calldata tokens) external protected {
+        ITokenHolder feeWallet = networkFeeWallet();
+
+        // retrieve conversion paths and the amounts to burn
+        (address[][] memory paths, uint256[] memory amounts, address[] memory govPath) =
+            vortexStrategy(tokens, feeWallet);
+
+        // withdraw all token/ETH amounts to the contract
+        feeWallet.withdrawTokensMultiple(tokens, address(this), amounts);
+
+        // convert all amounts to the network token and record conversion amounts
+        BancorNetwork network = bancorNetwork();
+
+        uint256[] memory networkTokenConversionAmounts = new uint256[](tokens.length);
+        uint256 grossNetworkTokenConversionAmount = 0;
+        uint256 totalGovTokenAmountToBurn = 0;
+
+        for (uint256 i = 0; i < paths.length; ++i) {
+            // avoid empty conversions
+            uint256 amount = amounts[i];
+            if (amount == 0) {
+                continue;
+            }
+
+            address[] memory path = paths[i];
+            IERC20 token = IERC20(path[0]);
+            uint256 value = 0;
+
+            if (token == _govToken) {
+                // if the source token is the governance token, don't try to convert it either, but rather include it in
+                // the amount to burn
+                //
+                // Please note that networkTokenConversionAmounts[i] will remain 0 in this case
+                totalGovTokenAmountToBurn = totalGovTokenAmountToBurn.add(amount);
+            } else if (token == _networkToken) {
+                // if the source token is the network token, don't try to convert it, but rather add its amount to the
+                // total amount to convert to the governance token
+                grossNetworkTokenConversionAmount = grossNetworkTokenConversionAmount.add(amount);
+
+                networkTokenConversionAmounts[i] = amount;
+            } else {
+                if (token != NATIVE_TOKEN_ADDRESS) {
+                    // if the source token is a regular token, approve the converter to withdraw the token amount
+                    ensureAllowance(token, network, amount);
+                } else {
+                    // if the source token is actually an ETH reserve, make sure to pass its value to the network
+                    value = amount;
+                }
+
+                // perform the actual conversion and optionally send ETH to the network
+                uint256 networkTokenConversionAmount =
+                    network.convertByPath2{ value: value }(path, amount, 1, address(this));
```

Comment:
So min-return does not protect us here (and we obviously don't want to ask the user to provide it per token).
How can we be sure that we're protected from front-running here?

Reply:
Yes, this is by design, like we've discussed during the call. Allowing the caller to arb the conversion is part of the incentive mechanism